specs/agents/tracing-instrumentation-aws-lambda.md

```diff
@@ -132,6 +132,66 @@ In version 2.0, the `${event.requestContext.routeKey}` can have the format `GET
 
 If `use_path_as_transaction_name` is applicable and set to `true`, use `${event.requestContext.http.method} ${event.requestContext.http.path}` as the transaction name.
 
+### Elastic Load Balancer (ELB)
+
+Elastic Load Balancer (ELB) can be attached directly to lambda, without the use
+of API Gateway. In this case the `event` object will be structured differently.
+
+The agent should use the information in the request and response objects to
+fill the HTTP context (`context.request` and `context.response`) fields in the
+same way it is done for HTTP transactions.
+
+In particular, agents must use the `event.headers` to retrieve the
+`traceparent` and the `tracestate` and use them to start the transaction for
+the lambda function execution.
+
+In addition the following fields should be set for ELB-based Lambda functions:
+
+Field | Value | Description | Source
+---   | ---   | ---         | ---
+`type` | `request`| Transaction type: constant value for ELB. | -
+`name` | e.g. `GET /prod/proxy/{proxy+}` | Transaction name: Http method followed by a whitespace and the (resource) path. See section below. | -
+`transaction.result` | `HTTP Xxx` / `success` | `HTTP 5xx` if there was a function error (see [Lambda error handling doc](https://docs.aws.amazon.com/lambda/latest/dg/services-apigateway.html#services-apigateway-errors). If the [invocation response has a "statusCode" field](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.response), then set to `HTTP Xxx` based on the status code, otherwise `success`. | Error or `response.statusCode`.
+`faas.trigger.type` | `http` | Constant value for ELB. | -
+`faas.trigger.request_id` | e.g. `Root=1-xxxxxxxxxxxxxx` | AWS Trace ID of the ELB request. | `event.headers.x-amzn-trace-id`
```

Comment:
The `X-Amzn-Trace-Id` header is described here: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-request-tracing.html
FWIW it can include data from the caller, e.g. `Self=1-67891234-12456789abcdef012345678;Root=1-67891233-abcdef012345678912345678;CalledFrom=app`.  I don't know of a reason that would be a problem for us using it for `faas.trigger.request_id`. In other words, it sounds reasonable to use the header for `request_id`. I suppose we don't want to get into parsing it to get just the "id" part of the "Root=" or "Self=" segment.

I think it would be worth showing an actual example rather than `Root=1-xxxxxxxxxxxxxx` -- unless the width makes the rendered Markdown table unreadable -- and/or linking to the header doc.

Do we know if `event.headers.*` are all always lowercased?

Reply:
From https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html#receive-event-from-load-balancer it seems like the answer is yes, since they refer to the injected headers using mixed case, but in the event they are all lower.

Anyway, I added a "real" example.